### PR TITLE
return contract address for deploy account transaction receipt

### PR DESCRIPTION
**Pull request by marioiordanov**
## Usage related changes

getTransactionReceipt returns contract address for DeployAccountTransaction

## Development related changes

- Added function for returning ContractAddress
- Added integration test

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

**Review by ivpavici — APPROVED** [review]
lgtm

File: crates/starknet-server/tests/get_transaction_receipt_by_hash.rs [hunk 1]
```diff
@@ -18,11 +21,61 @@ mod get_transaction_receipt_by_hash_integration_tests {
...
+            devnet.get("/predeployed_accounts", None).await.unwrap();
+
+        let predeployed_accounts_json = get_json_body(predeployed_accounts_response).await;
+        let first_account = predeployed_accounts_json.as_array().unwrap().get(0).unwrap();
+
+        let private_key =
+            Felt::from_prefixed_hex_str(first_account["private_key"].as_str().unwrap()).unwrap();
+
+        // constructs starknet-rs account
+        let signer = LocalWallet::from(SigningKey::from_secret_scalar(private_key.into()));
```
Comment (FabijanC) [inline crates/starknet-server/tests/get_transaction_receipt_by_hash.rs:44]: Here we are basically using the data of an already predeployed account for a newly deployed account. I was confused by this. For tests involving new account deployment, we have a utility function called `get_deployable_account_signer` which returns a dummy undeployed value.

**Review by FabijanC — CHANGES_REQUESTED** [review]
Everything looks functionally correct, just left some minor improvement proposals.

File: crates/starknet-server/tests/get_transaction_receipt_by_hash.rs [hunk 2]
```diff
@@ -18,11 +21,61 @@ mod get_transaction_receipt_by_hash_integration_tests {
...
+            FieldElement::from_hex_be(CAIRO_0_ACCOUNT_CONTRACT_HASH).unwrap(),
+            CHAIN_ID,
+            signer.clone(),
+            devnet.clone_provider(),
+        )
+        .await
+        .unwrap();
+        let new_account_nonce = FieldElement::ZERO;
+
+        let deployment = account_factory.deploy(FieldElement::THREE);
```
[the remark this answers is not in the document]
Comment (marioiordanov) [reply crates/starknet-server/tests/get_transaction_receipt_by_hash.rs]: [df75a44](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/149/commits/df75a440b84bb814fa83af06a7e202e4df3b3f1d)

File: crates/starknet-server/tests/get_transaction_receipt_by_hash.rs (hunk 1 above)
Comment (marioiordanov) [reply crates/starknet-server/tests/get_transaction_receipt_by_hash.rs]: [3edcd76](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/149/commits/3edcd763592503fc256698434217b8cf3c4e9005)

**Review by FabijanC — APPROVED** [review]
Assuming you reran the test locally and that it passed, I'm approving.

**Pull request merged**